backend/app/api/endpoints/knowledge.py

```
@@ -468,6 +492,134 @@ def create_knowledge_base(
+            head_commit=data.head_commit,
+            changed_paths=(
+                None
+                if data.changed_paths is None
+                else [
+                    ChangedPath(path=item.path, status=item.change_type)
+                    for item in data.changed_paths
+                ]
+            ),
+        )
+    except GenerationInFlight as e:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)